R/check.R

```diff
@@ -285,21 +305,26 @@ check_calendar_timestep <- function(dates, date_var, exact = TRUE) {
 #' @return This function is used for its side effect of stopping if the check
 #' fails. If the check passes, the function returns invisibly.
 #' @family check
+#' @importFrom rlang abort
 check_numeric_timestep <- function(dates, date_var, timestep, exact = TRUE) {
   diffs <- as.numeric(
     difftime(dates[-1], dates[-length(dates)], units = "days")
   )
 
   if (any(diffs == 0)) {
-    stop(
-      date_var, " has a duplicate date. Please remove duplicate dates."
+    rlang::abort(
```

Comment:
Two related notes:
 - seems like doing abort => paste0 in some places, abort => paste in others. Need to pick one to minimize future developer surprise.
 - also seems like several abort => pastes - DRY opportunity?

Reply:
Then probably `paste0` is the more general solution

Reply:
I also like the helper function - the `...` behaviour of the base `stop()` is quite nice